src/backend/parser/parse_agg.c

```diff
@@ -878,11 +878,12 @@ parseCheckAggregates(ParseState *pstate, Query *qry)
 	}
 
 	/*
-	 * Getting adjacent TargetEntrys in groupings. Common tle will be needed
-	 * in case of ungrouped attributes in target list. In fact we get a list
-	 * of tle->ressortgroupref because we need to check for matching expressions
-	 * after flatten_joinalias_vars. We also find out if there are grouping
-	 * expressions.
+	 * Getting ressortgrouprefs of common TargetEntrys in groupings.
+	 * TargetEntry is common if it appears in several grouping expressions
```

Comment:
> "is common if it appears in several grouping expressions
	 * (including nesting) and in a regular GROUP BY (general case)."

Not if 
> "is common if it appears in all grouping expressions
	 * (including nesting) or in a regular GROUP BY (general case)."

?

Reply:
Then for the case of `GROUPING SETS(a)` I will point out that it degenerates into a regular GROUP BY.

Reply:
Done